### ISSUE TITLE
No Video Thumbnails

### ISSUE THREAD

**Issue by TheMohawkNinja** [ranger/ranger#2191]
#### Runtime Environment

- Operating system and version: Arch Linux 5.9.1
- Terminal emulator and version: rxvt-unicode v9.22
- Python version: 3.8.6
- Ranger version/commit: Commit 521d9f8
- Locale: en-US

#### Current Behavior

No thumbnail is displayed for videos (tested with .mp4 and .mkv formats). Not even the preview panel appears where the thumbnail should show.

Image previews do function properly, and ffmpegthumbnailer functions when called manually in the shell.

#### Expected Behavior

A thumbnail should appear when highlighting a video file.

#### Steps to reproduce
1. Set the following in ~/.config/ranger/rc.conf:
`set preview_script /home/user/.config/ranger/scope.sh`
`set preview_images true`
`set preview_images_method ueberzug` OR `set preview_images_method w3m`
`set show_hidden true`

2.  Uncomment the following lines in ~/.config/ranger/scope.sh
        video/*)
             ffmpegthumbnailer -i "${FILE_PATH}" -o "${IMAGE_CACHE_PATH}" -s 0 && exit 6
             exit 1;;
3. Open ranger and highlight a video file

**Comment by toonn** [comment]
What does `file --mimetype` report on those video files?

**Comment by qinohe** [comment]
I'd like to add that I can confirm the issue with this build.(521d9f8)
Though, the stable version on Arch Linux https://archlinux.org/packages/community/any/ranger/ Version: ranger-1.9.3-3 does work and shows a thumbnail, thanks

**Comment by TheMohawkNinja** [comment]
> What does `file --mimetype` report on those video files?

myvideo.mp4: video/mp4; charset=binary

**Comment by TheMohawkNinja** [comment]
> I'd like to add that I can confirm the issue with this build.([521d9f8](https://github.com/ranger/ranger/commit/521d9f89f5bc3572f54033d04f8ebac26ffa10c6))
> Though, the stable version on Arch Linux https://archlinux.org/packages/community/any/ranger/ Version: ranger-1.9.3-3 does work and shows a thumbnail, thanks

That's odd, I tried the latest stable version on Arch earlier today and the program just immediately crashed.

**Comment by qinohe** [comment]
I'm not sure why but it seems your python is from September version is now :python 3.9.1-1
Are you synced?

**Comment by TheMohawkNinja** [comment]
> I'm not sure why but it seems your python is from September version is now :python 3.9.1-1
> edit:which is probably the reason for ranger 1.9.3-3 was rebuild against python 3.9.1-1
> Are you synced?

What do you mean by "synced" exactly?

I manually installed python 3.9.1 and ran the latest ranger commit with it, and still no video thumbnails.

**Comment by qinohe** [comment]
> What do you mean by "synced" exactly?
>
Are you up date..
> I manually installed python 3.9.1 and ran the latest ranger commit with it, and still no video thumbnails.
>
This is a partial update and considered 'dangerous' unless you know what you're doing.
> EDIT: Also did a `pacman -Syyu` which looked like it pulled 3.9.1-1 from the package name, but python --version still only shows 3.9.1. Not sure if that's intentional.
>
No, need for that second refresh (y) it puts unnecessary load on arch servers, you would be noted in a news item to perform such update 
The '-1' is simply Arch packaging, try 'pacman -Q python'

**Comment by TheMohawkNinja** [comment]
> > What do you mean by "synced" exactly?
> 
> Are you up date..

Well, I'm running the latest commit of ranger on the latest release of Python

> > EDIT: Also did a `pacman -Syyu` which looked like it pulled 3.9.1-1 from the package name, but python --version still only shows 3.9.1. Not sure if that's intentional.
> 
> No, need for that second refresh (y) it puts unnecessary load on arch servers, you would be noted in a news item to perform such update

I wanted to make sure the mirrors were up-to-date. I've had issues in the past with pacman trying to pull old versions of software before because the mirror list was not correct.

> The '-1' is simply Arch packaging, try 'pacman -Q python'

Returned `python 3.9.1-1`

**Comment by toonn** [comment]
@qinohe, so you're saying v1.9.3 worked but 521d9f8 doesn't? That's bizarre. The full diff for `scope.sh` between those commits is:
```
toonn@cruithne ~/s/ranger> git diff v1.9.3 521d9f8 -- ranger/data/scope.sh
diff --git a/ranger/data/scope.sh b/ranger/data/scope.sh
index f403ed83..a3e1bedf 100755
--- a/ranger/data/scope.sh
+++ b/ranger/data/scope.sh
@@ -104,7 +104,7 @@ handle_extension() {
             ;;
 
         ## JSON
-        json)
+        json|ipynb)
             jq --color-output . "${FILE_PATH}" && exit 5
             python -m json.tool -- "${FILE_PATH}" && exit 5
             ;;
```

That hardly looks like it'd affect video previews, and it's a *very* recent change, last week or something.

Maybe what's happening is you don't have pillow installed for the new python or something? Some of the previewing methods depend on it.

**Comment by qinohe** [comment]
@toonn, thanks. Well pillow didn't make a change ( had it installed and removed it ), at least not for the video preview.
I was always using stable from Arch Linux and preview for video was 'always' working without using the thumbnailer case part in scope. Now for the recent version from git ( 521d9f8 ) I do need to enable the thumbnailer case and now it's working, 'mistery' solved, at least for me.
There was another difference, the git version shows a frame that looks like a celluloid film frame the stable ( 1.9.3 ) doesn't.

@TheMohawkNinja, I have a feeling your issue may be environmental and maybe don't belong here but on Arch bbs, just a thought...
You know there is a thread specially for ranger ( see )  https://bbs.archlinux.org/viewtopic.php?id=93025

**Comment by toonn** [comment]
I suspect arch simply patches ranger to enable those previews. (Why do so many packagers feel the need to customize default configurations? It's super confusing and creating a config file is a no-brainer with `--copy-config`.) Hence the difference between running a ranger installed via pacman and from the repo.
@TheMohawkNinja seems to be having a different issue altogether, they uncommented scope and enabled image previews in rc.conf.

**Comment by qinohe** [comment]
@toonn , I can assure you there's no patch for ranger ( see ) https://github.com/archlinux/svntogit-community/blob/packages/ranger/trunk/PKGBUILD and Arch is pretty 'vanilla' when it comes to hosting upstream, the user is in full control. Patches are only made if an adjustment needs to be made to make a program work on Arch infrastructure. So, I have no idea why both versions differ in regard to that. Though, I do agree it's confusing if downstream decides to 'think' for you. Thanks again  ;)

**Comment by toonn** [comment]
Then how can you have had video thumbnails by default? Manjaro is very annoying in this regard, they carry an ancient `scope.sh` patch.

**Comment by qinohe** [comment]
I decided to create new config files just to rule out the possibility of it being me ( ranger --copy-config=all )
Setting 'set preview_images true' seems to be enough to get a preview for a .avi/.mp4 ( tested ) on both 1.9.3 and 521d9f8.
This is the only thing I did change!
There was no need to enable thumbnailer in scope - for both!
The preview looks different on both; 1.9.3 shows a plain thumbnail , 521d9f8 shows a 'filmstrip/celluloid-like' thumbnail

**Comment by TheMohawkNinja** [comment]
> I decided to create new config files just to rule out the possibility of it being me ( ranger --copy-config=all )
> Setting 'set preview_images true' seems to be enough to get a preview for a .avi/.mp4 ( tested ) on both 1.9.3 and [521d9f8](https://github.com/ranger/ranger/commit/521d9f89f5bc3572f54033d04f8ebac26ffa10c6).
> This is the only thing I did change!
> There was no need to enable thumbnailer in scope - for both!

I backed up my scope.sh and rc.conf files and ran `ranger --copy-config=all` and ONLY changed `set preview_images` to  `true`.

Still the same result on both 1.9.3. and 521d9f8. Picture previews work, but no video thumbnails.

**Comment by qinohe** [comment]
@TheMohawkNinja , I'm pretty sure your issues are not within Ranger but is some environmental issue caused by settings you made on Arch, i3WM or some other setting but not ranger, sorry...

**Comment by TheMohawkNinja** [comment]
> @TheMohawkNinja , I'm pretty sure your issues are not within Ranger but is some environmental issue caused by settings you made on Arch, i3WM or some other setting but not ranger, sorry...

I'd be inclined to agree, but what could I even check exactly? The strangest part is that ffmpegthumbnailer works when called manually, so something is blocking ranger from triggering it.

**Comment by TheMohawkNinja** [comment]
Fixed the issue.

So, after putting in a few `echo` commands into scope.sh to confirm that yes, the `video/*)` section WAS being called, I ran ffmpegthumbnailer manually to find out that it was now throwing an error (`ffmpegthumbnailer: error while loading shared libraries: libffmpegthumbnailer.so.4: cannot open shared object file: No such file or directory`). While my previous `sudo pacman -Syyu` command should have pulled the latest version if available, I ran `sudo pacman -S ffmpegthumbnailer` which pulled a new version.

Video thumbnails now work.

I must have had multiple issues corroborate with themselves, because I was previously getting an error with ffmpegthumbnailer before (99% sure it was the exact same error), but I resolved it at the time and confirmed that it ran manually, so maybe `sudo pacman -Syyu` pulled an old version somehow and re-broke ffmpegthumbnailer?

Well, at least it works now.
